src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -684,36 +653,61 @@ private static ISymbol GetEventAccessor(IEventSymbol iEvent, IOperation operatio
                 {
-                    CheckOperationAttributes(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, accessor);
-                }
-                else
-                {
-                    CheckOperationAttributes(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, iEvent);
+                    CheckOperationAttributes(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, accessor, checkParents: false);
                 }
             }
-            else
+            else if (symbol is IMethodSymbol method && method.IsGenericMethod)
             {
-                CheckOperationAttributes(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, symbol);
+                CheckTypeArguments(method.TypeArguments, operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms);
             }
 
-            static void CheckOperationAttributes(IOperation operation, OperationAnalysisContext context, PooledConcurrentDictionary<IOperation,
-                SmallDictionary<string, PlatformAttributes>> platformSpecificOperations,
-                ConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>?> platformSpecificMembers, ImmutableArray<string> msBuildPlatforms, ISymbol symbol)
+            if (symbol.ContainingSymbol is INamedTypeSymbol namedType && namedType.IsGenericType)
             {
-                if (TryGetOrCreatePlatformAttributes(symbol, platformSpecificMembers, out var operationAttributes))
+                CheckTypeArguments(namedType.TypeArguments, operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms);
+            }
+
+            static void CheckTypeArguments(ImmutableArray<ITypeSymbol> typeArguments, IOperation operation, OperationAnalysisContext context,
+                PooledConcurrentDictionary<KeyValuePair<IOperation, ISymbol>, SmallDictionary<string, PlatformAttributes>> platformSpecificOperations,
+                ConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>?> platformSpecificMembers, ImmutableArray<string> msBuildPlatforms)
+            {
+                foreach (var typeArgument in typeArguments)
                 {
-                    if (TryGetOrCreatePlatformAttributes(context.ContainingSymbol, platformSpecificMembers, out var callSiteAttributes))
+                    if (typeArgument.SpecialType == SpecialType.None)
+                    {
+                        CheckOperationAttributes(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, typeArgument, checkParents: false);
+
+                        if (typeArgument is INamedTypeSymbol nType && nType.IsGenericType)
+                        {
+                            foreach (var tArgument in nType.TypeArguments)
```

Comment:
Instead of this for loop, should we just make a recursive call to `CheckTypeArguments` here? This will ensure things like `GenericType<SomeType<WindowsOnlyType>>.NonGenericMethod();` get flagged where the violating type is a nested type argument. I think to avoid infinite recursion, we probably need to maintain a worklist/set of types currently being processed by `CheckTypeArguments` so we don't repeatedly call it for the same type when we have broken code, say `GenericType<SomeType<GenericType>>.NonGenericMethod();`?

Reply:
Sure, this PR change is causing lots of new warnings in runtime, i am digging into that since yesterday, will apply your suggestion after that